src/components/SetupApplication/SetupApplicationForm.tsx

```diff
@@ -0,0 +1,259 @@
+            formState.values.openYear,
+            formState.values.openMonth - 1,
+            formState.values.openDay
+        );
+        if (
+            openDate.getFullYear() !== formState.values.openYear ||
+            openDate.getMonth() !== formState.values.openMonth + 1 ||
+            openDate.getDate() !== formState.values.openDay
+        ) {
+            isValidSoFar = false;
+        }
+
+        const closeDate = new Date();
+        closeDate.setFullYear(
+            formState.values.closeYear,
+            formState.values.closeMonth - 1,
+            formState.values.closeDay
+        );
+        if (
+            (isValidSoFar === true &&
+                closeDate.getFullYear() !== formState.values.closeYear) ||
+            closeDate.getMonth() !== formState.values.closeMonth + 1 ||
+            closeDate.getDate() !== formState.values.closeDay
+        ) {
+            isValidSoFar = false;
+        }
+
+        Object.keys(formState.errors).length && !isValidSoFar
+            ? setValidForm(false)
+            : setValidForm(true);
+    };
+
+    useEffect(() => {
+        checkForErrors();
+    }, [formState]);
+
+    function handleSubmit(event: SyntheticEvent) {
+        event.preventDefault();
+
+        let year = formState.values.openYear;
+        let month = formState.values.openMonth;
+        let day = formState.values.openDay;
+        let hour = formState.values.openHour;
+        let minute = formState.values.openMinute;
+
+        const openDate = new Date(year, month, day, hour, minute).toISOString();
+
+        year = formState.values.closeYear;
+        month = formState.values.closeMonth;
+        day = formState.values.closeDay;
+        hour = formState.values.closeHour;
+        minute = formState.values.closeMinute;
+
+        const closeDate = new Date(
+            year,
+            month,
+            day,
+            hour,
+            minute
+        ).toISOString();
+
+        updateApplication(openDate, closeDate);
+    }
+
+    function validateDay(value: string) {
+        const intValue = parseInt(value);
+        if (!value || isNaN(intValue) || intValue > 31) {
+            return "Incorrect";
+        }
+    }
+
+    function validateYear(value: string) {
+        const intValue = parseInt(value);
+        if (!value || isNaN(intValue) || intValue > 2040 || intValue < 2019) {
+            return "Incorrect";
+        }
+    }
+
+    function validateMonth(value: string) {
+        const intValue = parseInt(value);
+        if (!value || isNaN(intValue) || intValue > 12) {
+            return "Incorrect";
+        }
+    }
+
+    function validateHour(value: string) {
+        const intValue = parseInt(value);
+        if (!value || isNaN(intValue) || intValue > 23) {
+            return "Incorrect";
+        }
+    }
+
+    function validateMinute(value: string) {
+        const intValue = parseInt(value);
+        if (!value || isNaN(intValue) || intValue > 59) {
```

Comment:
Won't this fail on minute zero?

Reply:
It was ok here, it was further down the line when populating the page with 00's for the start of the hour.